src/containers/pages/InterventionPlan/IRS/report/index.tsx

```diff
@@ -0,0 +1,610 @@
+import plansReducer, {
+  getPlanRecordById,
+  PlanRecord,
+  reducerName as plansReducerName,
+} from '../../../../../store/ducks/plans';
+
+import { TableCrumb } from '../plan';
+import './style.css';
+/** register the plans reducer */
+reducerRegistry.register(jurisdictionReducerName, jurisdictionReducer);
+reducerRegistry.register(plansReducerName, plansReducer);
+
+/** interface to describe props for IrsReport component */
+export interface IrsReportProps {
+  childrenByParentId: ChildrenByParentId;
+  fetchChildrenByParentIdActionCreator: typeof fetchChildrenByParentId;
+  fetchJurisdictionIdsByPlanIdActionCreator: typeof fetchJurisdictionIdsByPlanId;
+  fetchJurisdictionsActionCreator: typeof fetchJurisdictions;
+  jurisdictionIdsByPlanId: JurisdictionIdsByPlanId;
+  jurisdictionsById: { [key: string]: Jurisdiction };
+  planById: PlanRecord | null;
+  planId: string;
+  supersetService: typeof supersetFetch;
+}
+
+/** default props for IrsReport component */
+export const defaultIrsReportProps: IrsReportProps = {
+  childrenByParentId: {},
+  fetchChildrenByParentIdActionCreator: fetchChildrenByParentId,
+  fetchJurisdictionIdsByPlanIdActionCreator: fetchJurisdictionIdsByPlanId,
+  fetchJurisdictionsActionCreator: fetchJurisdictions,
+  jurisdictionIdsByPlanId: {},
+  jurisdictionsById: {},
+  planById: null,
+  planId: '',
+  supersetService: supersetFetch,
+};
+
+export interface IrsReportState {
+  childlessChildrenIds: string[];
+  childrenByParentId: { [key: string]: string[] };
+  country: JurisdictionsByCountry | null;
+  doRenderTable: boolean;
+  filteredJurisdictionIds: string[];
+  focusJurisdictionId: string | null;
+  isLoadingHierarchy: boolean;
+  isLoadingPlanData: boolean;
+  reportTableProps: DrillDownProps<any> | null;
+  tableCrumbs: TableCrumb[];
+}
+
+export const defaultIrsReportState: IrsReportState = {
+  childlessChildrenIds: [],
+  childrenByParentId: {},
+  country: null,
+  doRenderTable: false,
+  filteredJurisdictionIds: [],
+  focusJurisdictionId: null,
+  isLoadingHierarchy: true,
+  isLoadingPlanData: false,
+  reportTableProps: null,
+  tableCrumbs: [],
+};
+
+/** Reporting for Single Active IRS Plan */
+class IrsReport extends React.Component<RouteComponentProps<RouteParams> & IrsReportProps, {}> {
+  public static defaultProps = defaultIrsReportProps;
+  public state = defaultIrsReportState;
+
+  public async componentDidMount() {
+    const {
+      fetchChildrenByParentIdActionCreator,
+      fetchJurisdictionIdsByPlanIdActionCreator,
+      fetchJurisdictionsActionCreator,
+      jurisdictionIdsByPlanId,
+      jurisdictionsById: JurisdictionsById,
+      planById,
+      planId,
+      supersetService,
+    } = this.props;
+
+    // get Jurisdicitons from Store or Superset
+    const jurisdictionsArray = Object.keys(JurisdictionsById).length
+      ? Object.keys(JurisdictionsById).map(j => JurisdictionsById[j])
+      : await supersetService(SUPERSET_IRS_REPORTING_JURISDICTIONS_DATA_SLICE, {
```

Comment:
Only load data if needed, might need to add some sort of checker helper to dynamically check if the jurisdictions need to be loaded based on IRS-type-specific conditions.